services/web/server/src/simcore_service_webserver/products_events.py

```diff
@@ -33,6 +36,27 @@ async def setup_product_templates(app: web.Application):
         # cleanup
 
 
+async def auto_create_products_groups(app: web.Application) -> int:
+    """Ensures all products have associated group ids
+
+    Avoids having undefined groups in products with new products.group_id column
+
+    NOTE: could not add this in 'setup_groups' (groups plugin)
+    since it has to be executed BEFORE 'load_products_on_startup'
+    """
+    engine: Engine = app[APP_DB_ENGINE_KEY]
+
+    async with engine.acquire() as conn:
+        async for row in iter_products(conn):
+            product_name = row.name
+            product_group_id = await get_or_create_product_group(conn, product_name)
+            log.debug(
+                "Missing group for %s, created %s",
+                f"{product_name=}",
+                f"{product_group_id=}",
+            )
+
```

Comment:
Please consider this might run in parallel, since we have multiple instances of the webserver. Can this cause issues?

Reply:
Good point! I made some changes in ``get_or_create_product_group `` to achieve this and added a test